airflow_kubernetes_job_operator/kubernetes_job_operator.py

```diff
@@ -270,6 +299,12 @@ def _update_override_params(self, o: dict):
                 self._update_main_container_yaml(containers[0])
 
                 # adding env resources
+            if self.volume_files_from_secret:
```

Comment:
This should be applied before the main container spec is applied, so future commands/derived commands can search for this. Please also add a comment.